bin/subtract_variants.py

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env python
+
+# Copyright 2024 - GHGA
+# Author: Kuebra Narci - @kubranarci
+
+import argparse
+import sys
+import gzip
+import bisect
+
+def open_file(file_path, mode='r'):
+    if file_path.endswith('.gz'):
+        return gzip.open(file_path, f'{mode}t')
+    else:
+        return open(file_path, mode)
+
+def parse_vcf(file_path):
+    variants = set()
+    try:
+        with open_file(file_path) as vcf_file:
+            for line in vcf_file:
+                if line.startswith('#'):
+                    continue
+                parts = line.strip().split('\t')
+                if len(parts) >= 5:
+                    variant_id = (parts[0], parts[1], parts[3], parts[4])
+                    variants.add(variant_id)
+    except FileNotFoundError:
+        print(f"Error: File not found at {file_path}", file=sys.stderr)
+        sys.exit(1)
+    except Exception as e:
+        print(f"An error occurred while parsing {file_path}: {e}", file=sys.stderr)
+        sys.exit(1)
+    return variants
+
+def parse_bed_file(file_path):
+    regions = {}
+    try:
+        with open_file(file_path) as bed_file:
+            for line in bed_file:
+                parts = line.strip().split('\t')
+                if len(parts) >= 3:
+                    chrom, start, end = parts[0], int(parts[1]), int(parts[2])
+                    if chrom not in regions:
+                        regions[chrom] = []
+                    regions[chrom].append((start, end))
+
+        for chrom in regions:
+            regions[chrom].sort(key=lambda x: x[0])
+
+    except FileNotFoundError:
+        print(f"Error: BED file not found at {file_path}", file=sys.stderr)
+        sys.exit(1)
+    except Exception as e:
+        print(f"An error occurred while parsing {file_path}: {e}", file=sys.stderr)
+        sys.exit(1)
+    return regions
+
+def is_in_region(chrom, pos, bed_regions):
+    if chrom not in bed_regions:
+        return False
+
+    chrom_regions = bed_regions[chrom]
+    idx = bisect.bisect_left(chrom_regions, (pos - 1, float('inf')))
+
+    if idx < len(chrom_regions):
+        start, end = chrom_regions[idx]
+        if pos >= start + 1 and pos <= end:
```

Comment:
MAybe good to check if BED follows standard format of 0 indexing. Meaning if the bed indixeing does not start with 0 then this might be wrong here.

Reply:
Thanks for the suggestion Flo, I added this to do list https://github.com/nf-core/variantbenchmarking/issues/256, we havent encounter yet but this may cause a bug really.